src/ex4/server/controllers/itemController.js

```diff
@@ -0,0 +1,143 @@
+const { v4: idKeyGen } = require("uuid");
+const { validate: uuidValidate } = require("uuid");
+const itemManagerService = require("../services/itemManagerService");
+const parserService = require("../services/parserService");
+const pokemonService = require("../services/pokemonClientService");
+
+async function createItem(req, res) {
+  //validation
+
+  const currentData = await itemManagerService.readItemFile();
+
+  const pokemonOrTaskResults = parserService.parseInputValue(
+    req.body.data.input
```

Comment:
This sort of concatenation usually requires validation because you assume and put your trust on too many things here:
- that you have body
- that body has .data
- that data has .input

if any of them will be undefined the entire code will crash